docs/adr/0005-multitenancy-approach-for-marain.md

```diff
@@ -0,0 +1,286 @@
+       |     |     +-> (FooBar storage configuration)
+       |     |
+       |     +-> WORKFLOW+Litware
+       |           +-> (Operations storage configuration)
+       |           +-> (The Id of the OPERATIONS+WORKFLOW+Litware sub-tenant for the Operations service to use)
+       |           +-> (FooBar storage configuration)
+       |
+       +-> OPERATIONS
+       |     |
+       |     +-> OPERATIONS+WORKFLOW+Contoso
+       |     |     +-> (FooBar storage configuration)
+       |     |
+       |     +-> OPERATIONS+WORKFLOW+Litware
+       |           +-> (FooBar storage configuration)
+       |
+       +-> FOOBAR
+```
+
+Since Litware is not licenced to use FooBar, the Litware Client Tenant does not hold any configuration for that service itself.
+
+Finally, we enroll Litware to use the Operations service. In this example, because Operations depends on FooBar, we need to create another sub-tenant of Operations to call FooBar with when Litware uses Operations directly, and enroll this new subtenant with FooBar. This leaves us with the following:
+
+```
+Root tenant
+ |
+ +-> Client Tenants
+ |     |
+ |     +-> Contoso
+ |     |     +-> (Workflow storage configuration)
+ |     |     +-> (The Id of the WORKFLOW+Contoso sub-tenant for the Workflow service to use)
+ |     |     +-> (FooBar storage configuration)
+ |     |
+ |     +-> Litware
+ |           +-> (Workflow storage configuration)
+ |           +-> (The Id of the WORKFLOW+Litware sub-tenant for the Workflow service to use)
+ |           +-> (Operations storage configuration)
+ |           +-> (The Id of the OPERATIONS+Litware sub-tenant for the Operations service to use)
+ |      
+ +-> Service Tenants
+       |
+       +-> WORKFLOW
+       |     |
+       |     +-> WORKFLOW+Contoso
+       |     |     +-> (Operations storage configuration)
+       |     |     +-> (The Id of the OPERATIONS+WORKFLOW+Contoso sub-tenant for the Operations service to use)
+       |     |     +-> (FooBar storage configuration)
+       |     |
+       |     +-> WORKFLOW+Litware
+       |           +-> (Operations storage configuration)
+       |           +-> (The Id of the OPERATIONS+WORKFLOW+Litware sub-tenant for the Operations service to use)
+       |           +-> (FooBar storage configuration)
+       |
+       +-> OPERATIONS
+       |     |
+       |     +-> OPERATIONS+WORKFLOW+Contoso
+       |     |     +-> (FooBar storage configuration)
+       |     |
+       |     +-> OPERATIONS+WORKFLOW+Litware
+       |     |     +-> (FooBar storage configuration)
+       |     |
+       |     +-> OPERATIONS+Litware
+       |           +-> (FooBar storage configuration)
+       |
+       +-> FOOBAR
+```
+
+## Consequences
+
+It is expected that these sub-tenants are created and configured as part of a process of enrolling tenants for specific services they are licenced to use. The means by which this is expected to happen is covered in ADR 0006.
```

Comment:
Permalink?

Reply:
It doesn't exist yet!

Reply:
(Once this is merged back, I'll add a link to this ADR as part of the PR for 0006.)